### PR TITLE
Feature/restart dict

### PR DESCRIPTION
**Pull request by sudarsan1989**
A new dictionary for specifying what the copied files for restart (wave functions, velocities, positions etc) should be named

### CODE REVIEW

File: pyiron_base/objects/job/generic.py [hunk 1]
```diff
@@ -281,6 +281,19 @@ def restart_file_list(self, filenames):
             except AssertionError:
                 raise IOError("File: {} does not exist".format(f))
 
+    @property
+    def restart_file_dict(self):
+        if len(self._restart_file_list) > 0:
+            for f in self._restart_file_list:
+                actual_name = f.split("/")[-1]
```
Comment (jan-janssen) [inline pyiron_base/objects/job/generic.py]: This is dangerous on windows systems - instead use `os.path.basename()`

File: pyiron_base/objects/job/generic.py [hunk 2]
```diff
@@ -859,9 +877,14 @@ def _copy_restart_files(self):
             assert (os.path.isdir(self.working_directory))
         except AssertionError:
             raise ValueError("The working directory is not yet available to copy restart files")
+        import shutil
         for f in self.restart_file_list:
-            import shutil
-            shutil.copy(f, self.working_directory)
+            actual_name = f.split("/")[-1]
```
Comment (jan-janssen) [inline pyiron_base/objects/job/generic.py]: again - `os.path.basename()`

File: pyiron_base/objects/job/generic.py [hunk 3]
```diff
@@ -859,9 +877,14 @@ def _copy_restart_files(self):
             assert (os.path.isdir(self.working_directory))
         except AssertionError:
             raise ValueError("The working directory is not yet available to copy restart files")
+        import shutil
```
Comment (jan-janssen) [inline pyiron_base/objects/job/generic.py]: Should not we use the import statements in the beginning?

File: pyiron_base/objects/job/generic.py [hunk 4]
```diff
@@ -281,6 +281,19 @@ def restart_file_list(self, filenames):
             except AssertionError:
                 raise IOError("File: {} does not exist".format(f))
 
+    @property
+    def restart_file_dict(self):
+        if len(self._restart_file_list) > 0:
+            for f in self._restart_file_list:
```
Comment (jan-janssen) [inline pyiron_base/objects/job/generic.py]: Maybe as list comprehension: 
```
for actual_name in [os.path.basename(f) for f in self._restart_file_list]:
    if actual_name not in self._restart_file_dict.keys():
        self._restart_file_dict[actual_name] = actual_name
```

File: pyiron_base/objects/job/generic.py [hunk 5]
```diff
@@ -281,6 +281,19 @@ def restart_file_list(self, filenames):
...
+                actual_name = f.split("/")[-1]
+                if actual_name not in self._restart_file_dict.keys():
+                    self._restart_file_dict[actual_name] = actual_name
+        return self._restart_file_dict
+
+    @restart_file_dict.setter
+    def restart_file_dict(self, val):
+        self._restart_file_dict = val
```
Comment (jan-janssen) [inline pyiron_base/objects/job/generic.py]: maybe add `assert` statement to validate it is a dictionary or `isinstance`.

**Comment by jan-janssen** [comment]
What about the `copy()` functions ?

File: pyiron_base/objects/job/generic.py (hunk 1 above)
Comment (sudarsan1989) [reply pyiron_base/objects/job/generic.py]: https://github.com/pyiron/pyiron_base/pull/21/commits/37aae86f528f208fc8dd7712987950c95233d39c

File: pyiron_base/objects/job/generic.py (hunk 3 above)
Comment (sudarsan1989) [reply pyiron_base/objects/job/generic.py]: https://github.com/pyiron/pyiron_base/pull/21/commits/7666909a0c440c83d445afb5cb8c49ed6ab70fa4

File: pyiron_base/objects/job/generic.py (hunk 2 above)
Comment (sudarsan1989) [reply pyiron_base/objects/job/generic.py]: See above

File: pyiron_base/objects/job/generic.py (hunk 4 above)
Comment (sudarsan1989) [reply pyiron_base/objects/job/generic.py]: See https://github.com/pyiron/pyiron_base/pull/21/commits/660d79ca42f27940731296cfac7b43ed2ea9966d

File: pyiron_base/objects/job/generic.py (hunk 5 above)
Comment (sudarsan1989) [reply pyiron_base/objects/job/generic.py]: See https://github.com/pyiron/pyiron_base/pull/21/commits/4316c9e1bb51e088039b377932343f060bde705e

**Pull request merged**